lightclient/src/platform/wasm_platform.rs

```diff
@@ -142,17 +141,15 @@ impl PlatformRef for SubxtPlatform {
             }
         };
 
-        Box::pin(async move {
+        let socket_future = async move {
             tracing::debug!(target: LOG_TARGET, "Connecting to addr={addr}");
+            WasmSocket::new(addr.as_str())
+                .map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err.to_string()))
+        };
 
-            let socket = WasmSocket::new(addr.as_str()).map_err(|err| ConnectError {
-                message: format!("Failed to reach peer: {err}"),
-            })?;
-
-            Ok(super::wasm_helpers::Stream(
-                super::wasm_helpers::with_buffers::WithBuffers::new(socket),
-            ))
-        })
+        future::ready(super::wasm_helpers::Stream(with_buffers::WithBuffers::new(
+            Box::pin(socket_future),
```

Comment:
a little annoying to box twice i.e, both the future and the stream but yeye

Reply:
It looks like only one box to me on the future, and the stream isn't boxed? Maybe I haven't dug deeply enough!